strawberryfields/ops.py

```diff
@@ -1408,11 +1409,13 @@ def __init__(self, s=1):
     def _decompose(self, reg, **kwargs):
         s = self.p[0]
         r = pf.asinh(-s/2)
-        #theta = 0.5*pf.atan2(-1.0/pf.cosh(r), -pf.tanh(r))
+        #theta = 0.5 * pf.atan2(-1.0 / pf.cosh(r), -pf.tanh(r))
         # FIXME in sympy 1.4 atan2._eval_evalf() has a bug, it does not work with Symbol._eval_evalf().
-        # This is a partial workaround for s>0. When sympy is fixed (in version 1.5?), go back to using pf.atan2.
-        # If s<0 we would need to add pi/2 to theta, but since s is symbolic at this point, it'd be complicated.
-        theta = 0.5 * pf.atan(1 / pf.sinh(r))
+        # This is a workaround. When sympy is fixed (in version 1.5?), go back to using pf.atan2.
+        # If s<0 we need to add pi/2 to theta. If s==0, we need to avoid division by zero.
+        theta = pf.Piecewise((0.5 * pf.atan(1 / pf.sinh(r)), s > 0),
+                             (-np.pi/4, s == 0),
+                             (0.5 * pf.atan(1 / pf.sinh(r)) +np.pi/2, True))
```

Comment:
Should it be `-np.pi/2` instead?
If `s<0`, then `r>0`, meaning `y<0` and `x<0`. From [wikipedia](https://en.wikipedia.org/wiki/Inverse_trigonometric_functions), when both `x` and `y` are less than zero, we get `atan2(y,x)=atan(y/x)-pi`. Am I mistaken somehow?

Reply:
`theta` is only defined up to `n \pi`, that should account for the difference.